Source/Curves/ContinuousCurves/SMCurveClass.f90

```diff
@@ -404,169 +391,166 @@ FUNCTION ObjectiveFunction(self,t) RESULT(d)
          d = (x(1)-xTarget(1))**2 + (x(2)-xTarget(2))**2
       END FUNCTION ObjectiveFunction
 !
+!////////////////////////////////////////////////////////////////////////
+!
+      DOUBLE PRECISION FUNCTION distanceSquared(x,c,p,nHat)
+         IMPLICIT NONE
+         DOUBLE PRECISION :: x, p(3), z(3), nHat(3)
+         CLASS(SMCurve)   :: c
+         
+         z               = c % positionAt(x)
+         distanceSquared = DistanceSquaredBetweenPoints(z,p,nHat)
+      END FUNCTION distanceSquared
+!
+!////////////////////////////////////////////////////////////////////////
+!
+      DOUBLE PRECISION FUNCTION DistanceSquaredBetweenPoints(z,p,nHat)
+         USE ProgramGlobals, ONLY: directionPenalty
+         IMPLICIT NONE
+         DOUBLE PRECISION ::  p(3), z(3), nHat(3)
+         
+         DistanceSquaredBetweenPoints = (z(1) - p(1))**2 + (z(2) - p(2))**2 &
+                           - MIN((z(1)-p(1))*nHat(1) + (z(2) - p(2))*nHat(2),0.0d0)/directionPenalty
+      END FUNCTION DistanceSquaredBetweenPoints
```

Comment:
```suggestion
      REAL(KIND=RP) FUNCTION distanceSquared(x,c,p,nHat)
         IMPLICIT NONE
         REAL(KIND=RP)  :: x, p(3), z(3), nHat(3)
         CLASS(SMCurve) :: c
         
         z               = c % positionAt(x)
         distanceSquared = DistanceSquaredBetweenPoints(z,p,nHat)
      END FUNCTION distanceSquared
!
!////////////////////////////////////////////////////////////////////////
!
      REAL(KIND=RP) FUNCTION DistanceSquaredBetweenPoints(z,p,nHat)
         USE ProgramGlobals, ONLY: directionPenalty
         IMPLICIT NONE
         REAL(KIND=RP) ::  p(3), z(3), nHat(3)
         
         DistanceSquaredBetweenPoints = (z(1) - p(1))**2 + (z(2) - p(2))**2 &
                           - MIN((z(1)-p(1))*nHat(1) + (z(2) - p(2))*nHat(2),0.0_RP)/directionPenalty
      END FUNCTION DistanceSquaredBetweenPoints
```
For consistency?

Reply:
Yes. DP was used only because FMM used it, and I wanted it to break if quad precision or something else was used. So now without that code, we should use RP.